[iOS] Remove AGX related sandbox extension code

pvollan opened the pull request:
#### 80c2766bbe0906ee3c97fe3107f18087b2671fa9
<pre>
[iOS] Remove AGX related sandbox extension code
<a href="https://bugs.webkit.org/show_bug.cgi?id=242124">https://bugs.webkit.org/show_bug.cgi?id=242124</a>
&lt;rdar://96153598&gt;

Reviewed by Chris Dumez.

Remove AGX related sandbox extension code on newer versions of iOS.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.GPU.sb.in:
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::platformInitializeWebProcess):
* Source/WebKit/UIProcess/GPU/GPUProcessProxy.cpp:
(WebKit::GPUProcessProxy::GPUProcessProxy):

Canonical link: <a href="https://commits.webkit.org/251999@main">https://commits.webkit.org/251999@main</a>
</pre>

cdumez reviewed (approved):
LGTM

cdumez commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1301,3 +1301,9 @@
 #endif
+
+#if (PLATFORM(IOS) && __IPHONE_OS_VERSION_MIN_REQUIRED < 160000) \
+    || PLATFORM(WATCHOS) \
```
Little surprised WatchOS and TvOS don't have a version check. Is it really iOS specific?

pvollan replied:
That is a good point! This is not iOS specific, but I am not aware of which versions of watchOS and tvOS has the feature, so keeping the old behavior for now.

Thanks for reviewing!

pvollan committed:
[iOS] Remove AGX related sandbox extension code
https://bugs.webkit.org/show_bug.cgi?id=242124
<rdar://96153598>

Reviewed by Chris Dumez.

Remove AGX related sandbox extension code on newer versions of iOS.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.GPU.sb.in:
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::platformInitializeWebProcess):
* Source/WebKit/UIProcess/GPU/GPUProcessProxy.cpp:
(WebKit::GPUProcessProxy::GPUProcessProxy):

Canonical link: https://commits.webkit.org/251999@main
